packages/nimble-components/src/select/index.ts

```diff
@@ -39,61 +89,776 @@ export class Select extends FoundationSelect implements ErrorPattern {
+    public maxHeight = 0;
+
+    /**
+     * The component is collapsible when in single-selection mode with no size attribute.
+     *
+     * @internal
+     */
+    @volatile
+    public get collapsible(): boolean {
+        return !(this.multiple || typeof this.size === 'number');
+    }
+
+    private _value = '';
+    private forcedPosition = false;
+    private indexWhenOpened?: number;
+
+    // This intersection observer is to handle focus behavior for when the dropdown becomes
+    // visible.
+    private readonly regionElementIntersectionObserver: IntersectionObserver = new IntersectionObserver(
+        entries => {
+            if (
+                entries.length > 0
+                    && entries[entries.length - 1]!.intersectionRatio > 0
+            ) {
+                this.focusAndScrollOptionIntoView();
+            }
+        },
+        { threshold: 1.0, root: document }
+    );
+
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this.addEventListener('change', this.changeValueHandler);
+        this.addEventListener('contentchange', this.updateDisplayValue);
+        this.forcedPosition = !!this.positionAttribute;
+        this.regionElementIntersectionObserver.observe(this.region);
+        this.initializeOpenState();
+    }
+
+    public override disconnectedCallback(): void {
+        this.removeEventListener('change', this.changeValueHandler);
+        this.removeEventListener('contentchange', this.updateDisplayValue);
+        this.regionElementIntersectionObserver.unobserve(this.region);
+        super.disconnectedCallback();
+    }
+
+    /**
+     * The list of options. This mirrors FAST's override implementation for this
+     * member for the Combobox to support a filtered list in the dropdown.
+     *
+     * @public
+     * @remarks
+     * Overrides `Listbox.options`.
+     */
+    public override get options(): ListboxOption[] {
+        Observable.track(this, 'options');
+        return this.filteredOptions?.length
+            ? this.filteredOptions
+            : (this._options as ListOption[]);
+    }
+
+    public override set options(value: ListboxOption[]) {
+        this._options = value;
+        Observable.notify(this, 'options');
+    }
+
+    // NOTE: This is a copy of the parent implementation. When providing an override
+    //  for a property setter, you must also provide its corresponding getter.
+    public override get value(): string {
+        Observable.track(this, 'value');
+        return this._value;
+    }
+
+    // This is copied directly from FAST's implemention of its Select component, with
+    // one main difference: we use 'options' (the filtered set of options) vs '_options'.
+    // This is needed because while the dropdown is open the current 'selectedIndex' (set
+    // within this implementation) needs to be relative to the filtered options.
+    public override set value(next: string) {
+        const prev = `${this._value}`;
```

Comment:
Why is the wrapping template string needed?

```suggestion
        const prev = this._value;
```